k2/addons/k2-upgrade/upgrade/upgrader.py

```diff
@@ -0,0 +1,60 @@
+import logging
+from concurrent.futures import TimeoutError
+
+from zaf.extensions.extension import get_logger_name
+
+from upgrade import PERFORM_UPGRADE, UPGRADE_DEFAULT_TIMEOUT
+
+logger = logging.getLogger(get_logger_name('k2', 'upgrade'))
+
+
+class NoUpgradeTypes(Exception):
+    pass
+
+
+class MultipleUpgradeTypes(Exception):
+    pass
+
+
+class UpgradeFailed(Exception):
+    pass
+
+
+class Upgrader(object):
```

Comment:
Write the `Upgrader` as a component so that it can be used in other contexts than just the command.

Reply:
Right! I forgot about the component! I was working on an implementation of the upgrade interface and forgot to clean up and improve the interface before pushing :roll_eyes: